### PR TITLE
Added new devicelock capability

### PR DESCRIPTION
**Pull request by kristofmuhi**
new RDC feature capability documentation:

### Description
Exposing to customer the new setupDeviceLock capability.
Implemented in https://saucedev.atlassian.net/browse/MRE-423. 

### Motivation and Context
We have customer apps which requires devices to be setup with pincodes/certain security levels. 
We are gonna restrict the ability to setup pincodes manually in the settings app, so we are exposing a new capability called setupDeviceLock, where Saucelabs is setting up the device lock for the customers, so we can easily remove it.

### Types of changes
- [ ] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: docs/dev/test-configuration-options.md [hunk 1]
```diff
@@ -443,6 +443,26 @@ For virtual device mobile tests, the capability is `deviceOrientation`, but for
 ```
 
 ---
+
+
+---
+### `setupDeviceLock`
+<p><small>| BOOLEAN | <span className="sauceDBlue">Real Devices Only</span> |</small></p>
```
Comment (wswebcreation) [inline docs/dev/test-configuration-options.md:450]: @kristofmuhi 
I can imagine situations where people need to test an E2E flow and the passcode needs to be submitted during that flow. Can we also provide which passcode we set?

Comment (kristofmuhi) [reply docs/dev/test-configuration-options.md:450]: that could make sense, but not in the scope of this MR documentation for the current behavior. 

Didnt hear about this usecase so far from our current customer base who use keyguard for app launches.

File: docs/dev/test-configuration-options.md [hunk 2]
```diff
@@ -443,6 +443,26 @@ For virtual device mobile tests, the capability is `deviceOrientation`, but for
...
+
+Sets up the device pin code for the automated test session.  Valid values are `true` and `false`.
```
Comment (lysannep) [inline docs/dev/test-configuration-options.md:452]:
```suggestion
Sets up the device pin code for the automated test session. Valid values are `true` and `false`.
```

File: docs/dev/test-configuration-options.md (hunk 1 above)
Comment (wswebcreation) [reply docs/dev/test-configuration-options.md:450]: Why not, it's a few words extra, unless we randomize it

**Review by aoleszyn — APPROVED** [review]
Approved.

**Review by aoleszyn — APPROVED** [review]
Approved.

**Pull request merged**